lib/http/timeout/global.rb

```diff
@@ -35,29 +35,65 @@ def connect_ssl
         end
       end
 
-      # Read from the socket
-      def readpartial(size)
-        reset_timer
+      # NIO with exceptions
+      if RUBY_VERSION < "2.1.0"
```

Comment:
I was thinking about it but I'd rather just do Socketry really. I was starting work on it last night but things blew up :(

Need to do a bit more testing on a lot of this though